tap_pendo/streams.py

```diff
@@ -473,7 +481,8 @@ def lookback_window(self):
 
 class LazyAggregationStream(Stream):
     def send_request_get_results(self, req):
-        with session.send(req, stream=True) as resp:
+        # add timeout if 'request_timeout' param found in else use default value
+        with session.send(req, stream=True, timeout=self.config.get('request_timeout', REQUEST_TIMEOUT)) as resp:
```

Comment:
The `LazyAggregationStream` stream extends the `Stream` class which contains function named `request` (responsible for sending requests) and from this function the function `send_request_get_results` is called ([here](https://github.com/singer-io/tap-pendo/blob/8a9b41086c5854b2701c6f6585c7bf5e940eeb00/tap_pendo/streams.py#L271)).
So when the Timeout Error will be raised it will be caught in the 'request' function and it will backoff the request.